core/services/src/main/java/org/exoplatform/services/attachments/listener/UpdateContentPermissionsListener.java

```diff
@@ -61,24 +63,33 @@ public class UpdateContentPermissionsListener extends Listener<Map<String, Objec
 
   private static final String          IMAGE_SRC_REGEX      = "src=\"/portal/rest/images/?(.+)?\"";
 
-  public UpdateContentPermissionsListener(RepositoryService repositoryService, SessionProviderService sessionProviderService) {
+  public UpdateContentPermissionsListener(RepositoryService repositoryService, AttachmentStorage attachmentStorage) {
     this.repositoryService = repositoryService;
-    this.sessionProviderService = sessionProviderService;
+    this.attachmentStorage = attachmentStorage;
   }
 
   @Override
   public void onEvent(Event event) throws Exception {
+
+    // set system session provider to avoid SessionProviderService NullPointerException with spring context
+    SessionProvider sessionProvider = SessionProvider.createSystemProvider();
+    SessionProviderService sessionProviderService = ExoContainerContext.getService(SessionProviderService.class);
+    sessionProviderService.setSessionProvider(null, sessionProvider);
```

Comment:
- you can put this directly into the updateNodePermissions method
- why not keeping SessionProviderService injected in the constructor ? it already works well before

Reply:
Because it is always null when injected by the constructor, it is annotated with @DefinitionByType so it is not injected when using the Spring context. Additionally, we cannot include this part of code directly in the updateNodePermissions method due to the dependency of the SessionProviderService on the attachment storage!